PhotoStreamer/app/src/main/java/com/mtruehle/photostreamer/HttpHandler.java

```diff
@@ -0,0 +1,90 @@
+
+
+    public HttpHandler(Context context) {
+        queue = Volley.newRequestQueue(context);
+    }
+
+    public void imageSearch(String searchQuery, int offset, Callback callback) {
+        final Callback staticCallback = callback;
+        searchQuery = searchQuery.replaceAll(" ", "+");
+        String offsetString = Integer.toString(offset+1);
+        String URL = BASE_URL + "?key=" + API_KEY + "&cx=" + SEARCH_ENGINE_ID + "&q=" + searchQuery + "&start=" + offsetString + "&searchType=image";
+        JSONObject body = new JSONObject(); // not sure what this is for. will try to find out later.
+        JsonObjectRequest getRequest = new JsonObjectRequest(
+                Request.Method.GET,
+                URL,
+                body,
+                new Response.Listener<JSONObject>() {
+                    @Override
+                    public void onResponse(JSONObject response) {
+                        ArrayList<String> imageLinkList = handleJson(response);
+                        staticCallback.callback(imageLinkList, true);
+                    }
+                },
+                new Response.ErrorListener(){
+                    @Override
+                    public void onErrorResponse(VolleyError error) {
+                        ArrayList<String> emptyList = new ArrayList<String>(); // callback needs it; doesn't get used.
+                        staticCallback.callback(emptyList, false);
+                    }
+                }
+        );
+        queue.add(getRequest);
+    }
+
+    public ArrayList<String> handleJson(JSONObject input) {
+        ArrayList<String> imageUrls = new ArrayList<String>();
+
+        try {
+            JSONArray jArray = input.getJSONArray("items");
+            int size = jArray.length();
+            if (size > 10) {
+                size = 10; // if more than 10 results get returned, limits to the first ten. Shouldn't happen, but just in case.
+            }
+            for (int i = 0; i < size; i++) {
+                JSONObject currentJsonObject = jArray.getJSONObject(i);
+                try {
+                    String currentUrl = currentJsonObject.getString("link");
+                    imageUrls.add(currentUrl);
+                } catch (JSONException ex) {
+//                    String currentUrl = "PASS"; // flags the callback to skip this item.
+//                    imageUrls.add(currentUrl);
+                    Log.e("Error", ex.getMessage());
+                }
+            }
+        } catch (JSONException ex) {
+            Log.e("Error", ex.getMessage());
+            imageUrls.add("ERROR"); // flags the callback to ignore all items.
+            imageUrls.add("ERROR");
+            imageUrls.add("ERROR"); // probably unnecessary extra declarations.
+        }
+        Log.i("PRINTER", "Results retrieved: " + imageUrls.get(0) + "\t" + imageUrls.get(1) + "\t" + imageUrls.get(2) + "...");
```

Comment:
You may find it more useful to put the class/method that the log is origination from in the first parameter